spec/features/search_page_spec.rb

```diff
@@ -4,113 +4,71 @@
 
 RSpec.feature 'Search page', type: :feature, elasticsearch: true do
   scenario 'Displays a not found message when a search returns 0 results' do
-    query = 'interesting dataset'
-
-    search_for(query)
-
+    search_for('interesting dataset')
     expect(page).to have_css('h1', text: 'Search results')
     expect(page).to have_content("0 results found")
   end
 
   scenario 'Displays search results' do
-    dataset_title = 'A very interesting dataset'
-    query = 'interesting dataset'
-
-    dataset = DatasetBuilder.new
-                .with_title(dataset_title)
-                .build
-
+    dataset = build :dataset, title: 'a title'
     index(dataset)
-    search_for(query)
+    search_for('title')
 
     expect(page).to have_css('h1', text: 'Search results')
-    expect(page).to have_css('a', text: dataset_title)
+    expect(page).to have_css('a', text: 'title')
```

Comment:
This assertion is now very generic – the page should have an anchor containing the text "title". There is a risk this could pass for the wrong reason.

Are we asserting it should have a link to the search term — which is now "title" — or is it a sub match on the title of the dataset — which is now "a title" — we expect to be in the results?

Reply:
Fair enough - being concise was my aim.